ui/raidboss/data/06-ew/raid/p6s.ts

```diff
@@ -85,6 +102,235 @@ const triggerSet: TriggerSet<Data> = {
+            const idx = x + y;
+            if (unsafeMap[idx] === undefined)
+              return;
+            tetheredTiles.push(idx);
+          }
+        }
+
+        // modifiers used to calculate unsafeMap indexes to be removed for each type of tile
+        const relCrossTiles = new Int8Array([-30, -20, -10, -3, -2, -1, 1, 2, 3, 10, 20, 30]);
+        const relDiagonalTiles = new Int8Array([-33, -27, -22, -18, -11, -9, 9, 11, 18, 22, 27, 33]);
+
+        for (const effect of data.mapEffects) {
+          if (mapLookup[effect.location] === undefined)
+            return;
+
+          const startTile: number = mapLookup[effect.location]!;
+          const isTethered: boolean = tetheredTiles.includes(startTile);
+
+          if (unsafeMap[startTile] !== undefined)
+            delete safe[unsafeMap[startTile]![1]]; // delete tile where effect appears, as it will always be unsafe
+          if ((effect.flags === crossTileFlags && !isTethered) || (effect.flags === diagonalTileFlags && isTethered)) {
+            relCrossTiles.forEach((tileMod) => {
+              const deleteTile: number = startTile + tileMod;
+              if (unsafeMap[deleteTile] !== undefined)
+                delete safe[unsafeMap[deleteTile]![1]];
+            });
+          } else if ((effect.flags === diagonalTileFlags && !isTethered) || (effect.flags === crossTileFlags && isTethered)) {
+            relDiagonalTiles.forEach((tileMod) => {
+              const deleteTile: number = startTile + tileMod;
+              if (unsafeMap[deleteTile] !== undefined)
+                delete safe[unsafeMap[deleteTile]![1]];
+            });
+          } else {
+            return;
+          }
+        }
+
+        const safeTiles: string[] = Object.keys(safe);
+        const [safe0, safe1] = safeTiles;
+
+        if (safe0 === undefined)
+          return;
+
+        const poly6Map: { [l: string]: string } = {
+          insideNW: 'insideSE',
+          insideNE: 'insideSW',
+          insideSE: 'insideNW',
+          insideSW: 'insideNE',
+        };
+
+        switch (data.polyInstance) {
+          case 1: // four safe spots: two inside, two outside.  call only the inside ones.
+            if (safeTiles.length !== 4 || safe1 === undefined || output[safe0] === undefined || output[safe1] === undefined)
+              return;
+            return output.combo!({ dir1: output[safe0]!(), dir2: output[safe1]!() });
+          case 2: // one inside safe spot
+            if (safeTiles.length !== 1 || output[safe0] === undefined)
+              return;
+            return output.single!({ dir1: output[safe0]!() });
+          case 3: // two inside safe spots
+            if (safeTiles.length !== 2 || safe1 === undefined || output[safe0] === undefined || output[safe1] === undefined)
+              return;
+            return output.combo!({ dir1: output[safe0]!(), dir2: output[safe1]!() });
```

Comment:
Similar to case 1, I wonder if this one should also say "healer groups (NW/SE)" or something.

Reply:
> Similar to case 1, I wonder if this one should also say "healer groups (NW/SE)" or something.

Yep, totally agree.  Oddly though, as you say that,  I don't remember ever seeing a healer group alert during that mechanic, or during the final poly (poly8) where that too is combined with a healer group stacks.  I've noticed it a couple times, but keep forgetting to bring it up/do something about it.

Looking at the triggers and my logs, I _think_ it's because the P6S Exocleaver Healer Groups trigger sets data.secondExocleavers = true (so the trigger doesn't execute twice for the second exocleaver hit) - but that var is not ever reset, so the trigger doesn't refire on subsequent occurrences of the unholy dark mechanic.  So I think the fix would be to set data.secondExocleavers = false in the second trigger (P6S Exocleaver Move), so the initial trigger is able to fire again the next time the mechanic occurs.  Let me know if that sounds right.

Happy to fix this as part of modifying the exocleaver trigger anyway for this PR - but lmk if you prefer to do it as a separate PR, and I can submit that tonight after raid.

Reply:
Yeah, that's exactly what I was thinking originally.  Although, in the other comment you say this is too much information in one line, so maybe it needs to be split up still.

Reply:
> Yeah, that's exactly what I was thinking originally. Although, in the other comment you say this is too much information in one line, so maybe it needs to be split up still.

Totally fair - and yeah, I do think it should be split up.  So giving this more thought, and with where we've landed for the other poly calls, I think it makes sense to have this poly trigger do infoText with the safe tiles (as mapeffects render), and then do a separate trigger with alertText for the healer groups when boss applies unholy darkness (but excluding the first instance during poly1 with exocleavers, since we already have a trigger for that that includes the exos).  Realize this is a little different than what I said above, but (1) it avoids having to turn this whole trigger into a response trigger just to handle dual text for these mechanics, (2) allows the safe tiles to be called early to give people time before the healer group trigger fires, and (3) keeps it as consistent as possible with how we're handling most of the other polys (safe tiles in infotext, other mechs in alerttext).

Reply:
Yeah, sounds good.